client/my-sites/checkout/checkout-thank-you/redesign-v2/sections/Product.tsx

```diff
@@ -0,0 +1,21 @@
+import DomainsTransferredList from './product/DomainsTransferredList';
+import ProductPlan, { ProductPlanProps } from './product/ProductPlan';
+import type { ReceiptPurchase } from 'calypso/state/receipts/types';
+
+const Product = ( {
+	siteSlug,
+	primaryPurchase,
+	siteID,
+	purchases,
+	isBulkDomainTransfer,
+}: ProductPlanProps & { purchases: ReceiptPurchase[]; isBulkDomainTransfer: boolean } ) => {
+	if ( isBulkDomainTransfer ) {
```

Comment:
@candy02058912 I don't think having to propdrill what flow you're in down to this level is scalable. e.g passing `isBulkDomainTransfer` 

Can we leverage purchase type from the list of purchases to render this section?

Maybe we want to inline whatever DomainsTransferredList is doing and delete that component?

Reply:
You're right @lsl, I created[ this PR](https://github.com/Automattic/wp-calypso/pull/79730) to try to address some of these concerns!

Reply:
Ah sorry, this was some drive by comments regarding a separate project we have in the works, the boundaries and way things fit together is still somewhat in flux. 

pb5gDS-36j-p2